packages/itwin/property-grid/src/hooks/UseDisposable.ts

```diff
@@ -0,0 +1,29 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import type { IDisposable } from "@itwin/core-bentley";
+import { useEffect, useRef, useState } from "react";
+
+// TODO: Remove when `useDisposable` from `@itwin/core-react` works fine with React 18 strict mode
+/** @internal */
+export function useDisposable<TDisposable extends IDisposable>(factory: () => TDisposable): TDisposable {
```

Comment:
Could you copy this into AppUi?

Reply:
Yes, I have local changes made in AppUI repo. Will open PR soon as I made changes to some other hooks in AppUI too.

Reply:
https://github.com/iTwin/appui/pull/546/files